assets/character/scripts/character_action_water_soil.gd

```diff
@@ -1,26 +1,49 @@
 class_name CharacterActionWaterSoil extends CharacterAction
 
-var water_cell: Vector2i
+const WATER_INTERVAL = 0.3
+
+var timer: Timer
+var mouse_down: bool
 var nav_to_action: CharacterAction
+var water_cell: Vector2i:
+	set(cell):
+		water_cell = cell
+		if active and nav_to_action:
+			var character_cell = GameManager.current_zone.grid.get_cell_by_position(character.global_position)
+			if Vector2(water_cell).distance_to(character_cell) > 3:
```

Comment:
This work is already being done by the character's nav agent. You can just ask if you reach destination (there is a distance allowance you can set) or not.

Reply:
Ahh cool, please show me this when we're next on, i was hoping for exactly that